### ISSUE TITLE
function build_val_loader

### ISSUE THREAD

**Issue by ElijahZh** [Pointcept/Pointcept#299]
Hi, thank you for the awesome code! This project helps me a lot. However, I am confused about the function `build_val_loader` in the file "main/pointcept/engines/train.py":
line 267
`val_sampler = torch.utils.data.distributed.DistributedSampler(val_data)`
the sampler doesn't drop the last, so how to guarantee in the distributed mode, the dataset is evenly divided.
I refer the validation sampler code in [imagenet_example](https://github.com/pytorch/examples/blob/main/imagenet/main.py)
They drop out the last in distributed mode, and run separate for the drop out data with Subset. 
I really appreciate it if you can help me understand the difference.

**Comment by Gofinge** [comment]
Hi, no need to be evenly divided for point cloud data in Pointcept. The batch size and num points are combined to support dynamic shapes. Check here: https://github.com/Pointcept/Pointcept?tab=readme-ov-file#offset

**Comment by ElijahZh** [comment]
> Hi, no need to be evenly divided for point cloud data in Pointcept. The batch size and num points are combined to support dynamic shapes. Check here: https://github.com/Pointcept/Pointcept?tab=readme-ov-file#offset

Hi, I asked similar question on [this](https://discuss.pytorch.org/t/understanding-distributedsampler-and-dataloader-drop-last/206271). If the data size is not evenly divided for the sampler in ddp, with `drop_last=False`, then it pads the indexes to divide them. My understanding is that, in the example of the val_sampler, there are repeat indices to make it even. However, since the augmentation of GridSample,  with the same instance, the return point set is not the same one.

**Comment by Gofinge** [comment]
Oh, yes. Some point clouds might be padded as `drop_last=False.` In the final test script, we maintain a dictionary to prevent duplicate point clouds involved in metric computation. But for the validation process, there is no need to care too much about accuracy.

**Comment by ElijahZh** [comment]
> Oh, yes. Some point clouds might be padded as `drop_last=False.` In the final test script, we maintain a dictionary to prevent duplicate point clouds involved in metric computation. But for the validation process, there is no need to care too much about accuracy.

Thank you so much for the help.
